lang/test/org/partiql/lang/eval/EvaluatingCompilerDateTimeTests.kt

```diff
@@ -34,4 +45,197 @@ class EvaluatingCompilerDateTimeTests : EvaluatorTestBase() {
+            case("TIME (2) WITH TIME ZONE '12:59:59.135-05:30'", "12:59:59.14-05:30", TimeForTest(12, 59, 59, 140000000, 2, -330)),
+            case("TIME (2) WITH TIME ZONE '12:59:59.134-05:30'", "12:59:59.13-05:30", TimeForTest(12, 59, 59, 130000000, 2, -330)),
+            case("TIME '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) '01:01:12' IS TIME", "true"),
+            case("TIME WITH TIME ZONE '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) WITH TIME ZONE '01:01:12' IS TIME", "true"),
+            case("'01:01:12' IS TIME", "false")
+        )
+    }
+
+    private val RANDOM_GENERATOR = generateRandomSeed()
+
+    private fun generateRandomSeed() : Random {
+        val seed = Random.nextInt()
+        println("Randomly generated seed is ${seed}. Use this to reproduce failures in dev environment.")
+        return Random(seed)
+    }
+
+    data class TimeForTest(
+        val hour: Int,
+        val minute: Int,
+        val second: Int,
+        val nano: Int = 0,
+        val precision: Int,
+        val tz_minutes: Int? = null
+    ) {
+        fun expectedTimeString(withTimeZone: Boolean): String {
+            val timezoneMinutes = when(withTimeZone) {
+                true -> tz_minutes ?: ZoneOffset.systemDefault().rules.getOffset(Instant.now()).totalSeconds / SECONDS_PER_MINUTE
+                else -> null
+            }
+            return Time.of(hour, minute, second, nano, precision, timezoneMinutes).toString()
+        }
+
+        override fun toString(): String {
+            val hourStr = hour.toString().padStart(2, '0')
+            val minStr = minute.toString().padStart(2,'0')
+            val secStr = second.toString().padStart(2, '0')
+            val nanoStr = nano.toString().padStart(9, '0')
+            val timezoneStr = tz_minutes?.let { "" +
+                (if (it >= 0) "+" else "-") +
+                (it.absoluteValue / 60).toString().padStart(2, '0') +
+                ":" +
+                (it.absoluteValue % 60).toString().padStart(2, '0')
+            } ?: ""
+            return "$hourStr:$minStr:$secStr.$nanoStr$timezoneStr"
+        }
+    }
+
+    private fun Random.nextTime(withPrecision: Boolean = false, withTimezone: Boolean = false) : TimeForTest {
+        val hour = nextInt(24)
+        val minute = nextInt(60)
+        val second = nextInt(60)
+        val nano = nextInt(999999999)
+        val precision = if (withPrecision) {
+            nextInt(10)
+        } else {
+            val timeStr = Time.of(hour, minute, second, nano, 9).toString()
+            timeStr.split(".")[1].length
+        }
+        val timezoneMinutes = if (withTimezone) {
+            nextInt(-1080, 1081)
+        } else {
+            null
+        }
+        return TimeForTest(hour, minute, second, nano, precision, timezoneMinutes)
+    }
+
+    private val RANDOM_TIMES = List(RANDOM_TESTS_SIZE) {
```

Comment:
Nit:  also move to top level in file, make `const` if possible.  My IntelliJ wants me to change the naming convention of this field, does it not for you?

Reply:
Changed to `camelCase`.